libraries/AP_HAL_SITL/UARTDriver.cpp

```diff
@@ -180,14 +180,23 @@ uint32_t UARTDriver::txspace(void)
 
 int16_t UARTDriver::read(void)
 {
-    if (available() <= 0) {
+    uint8_t c;
+    if (read(&c, 1) == 0) {
         return -1;
     }
-    uint8_t c;
-    _readbuffer.read(&c, 1);
     return c;
 }
 
+ssize_t UARTDriver::read(uint8_t *buffer, uint16_t count)
+{
+    const uint32_t ret = _readbuffer.read(buffer, count);
+    if (ret == 0) {
+        return 0;
+    }
```

Comment:
I do not get these three lines
```suggestion
```

Reply:
Did you meant to return -1?

Reply:
No - this is what the ChibiOS equivalent does.

0 is "I have no bytes to give you at the moment, which is fine, I think...

Reply:
Then ret == 0 and you are already returning it one line bellow. The if is redundant

Reply:
True enough, but the PR I extracted this from actually adds lines in that gap :-)

Reply:
@amilcarlucas I've taken the hint here and removed the extra lines.  I'll re-add them on the follow-up PR.